frontend/subrequests/lint/lint.go

```diff
@@ -0,0 +1,153 @@
+package lint
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/frontend/subrequests"
+)
+
+const RequestLint = "frontend.lint"
+
+var SubrequestLintDefinition = subrequests.Request{
+	Name:        RequestLint,
+	Version:     "1.0.0",
+	Type:        subrequests.TypeRPC,
+	Description: "Lint a Dockerfile",
+	Opts:        []subrequests.Named{},
+	Metadata: []subrequests.Named{
+		{Name: "result.json"},
+		{Name: "result.txt"},
+	},
+}
+
+type Source struct {
+	Filename string `json:"filename"`
+	Language string `json:"language"`
+	Data     []byte `json:"data"`
+}
+
+type Range struct {
```

Comment:
`outline` subrequest seems to use `pb.Location` type directly. Not sure if it is a good idea to have this dependency but atm you can do the same. If needed we can make some separate package later for these shared types that all subrequests can depend on.